### PR TITLE
Update Pulumi Managed Backend and Self-managed backend

### PR DESCRIPTION
**Pull request by tusharshahrs**
### Proposed changes

Updated the benefits of Pulumi backend vs self hosted backend to make it easier for customers/sales to make a decision.
Added additional plus points to Pulumi managed back end
Added risks that customers need to address for self-managed backend

### Related issues (optional)

 Related to: https://github.com/pulumi/docs/issues/4325
 Based on following community slack [discussion](https://pulumi-community.slack.com/archives/C019YSXN04B/p1601212645020300?thread_ts=1601061499.018300&cid=C019YSXN04B)

### CODE REVIEW

File: content/docs/intro/concepts/state.md [hunk 1]
```diff
@@ -99,6 +99,17 @@ At a glance, the Pulumi Service backend provides the following benefits:
...
+- Refresh config from the last deployment
+- Secrets handled by a key managed by the service or the option to use passpharse-based key
+
+### Self-managed backend risks
+
+At a glance, the self-hosted backend requires that the following issues are addressed:
+
+- Implement your own state management including backup, sharing, and team access synchronization
+- Automated deployments colliding and potentially corrupting state/making simultaneous edits to your infrastructure because there is no concurrency control
+- Secrets handled by a key that must use the passphrase-based secrets provider
```
Comment (stack72) [inline content/docs/intro/concepts/state.md:112]: This isn't true - we can use any of the non-SaaS based secret providers right?

File: content/docs/intro/concepts/state.md [hunk 2]
```diff
@@ -99,6 +99,17 @@ At a glance, the Pulumi Service backend provides the following benefits:
 - Full deployment history for auditing and rollback purposes
 - Concurrent state locking to prevent corrupting your infrastructure state in a team environment
 - Advanced policy and RBAC (Role Based Access Control)
+- Stack access control, including information on who owns a stack, who last modified stack, and what time the stack was modified
+- Refresh config from the last deployment
+- Secrets handled by a key managed by the service or the option to use passpharse-based key
+
+### Self-managed backend risks
```
Comment (stack72) [inline content/docs/intro/concepts/state.md:106]: I am not a fan of using the word "risks" here - they are not risks they just requirement that we need to ensure people understand the differences

**Review by lukehoban — COMMENTED** [review]
I had written up a much longer set of notes - but looks like they got lost.  

High level - I think unfortunately a lot of what's added here isn't correct, and we'll want to make sure that we rewrite to get the details right.  I'd also like to make an effort to reframe in the positive instead of negative - highlighting with helpful clarity what users would need to do to use the self-managed backends effectively (even if that does mean doing a fair bit more heavy lifting).  @stack72 Think you could help @tusharshahrs on the details and framing here?

**Comment by clstokes** [comment]
I think there are some other changes planned to this documentation, so I propose we close this PR and wait for those. @tusharshahrs, if there are some smaller changes you want to get in here, let's chat about those.

**Pull request closed without merging**